src/main/java/edu/bu/metcs673/trackr/user/TrackrUserServiceImpl.java

```diff
@@ -103,6 +103,7 @@ public TrackrUserDTO updateUser(TrackrUserDTO dto) {
             if (!bCryptPasswordEncoder.matches(dto.getPassword(), trackrUser.getPassword())) {
```

Comment:
Line 100 still has the the issue. This is not fixing it. How did you test? Please see below.
```java
@Override
    public TrackrUserDTO updateUser(TrackrUserDTO dto) {
        TrackrUser trackrUser = getCurrentUser();

        if (StringUtils.isNotBlank(dto.getNewPassword())) {
            regexPasswordValidation(dto.getPassword());
            if (!bCryptPasswordEncoder.matches(dto.getPassword(), trackrUser.getPassword())) {
                throw new TrackrInputValidationException(CommonConstants.WRONG_PASSWORD);
            } else {
                trackrUser.setPassword(bCryptPasswordEncoder.encode(dto.getNewPassword()));
            }
        }
```